include/libtorrent/alert_types.hpp

```diff
@@ -782,18 +782,23 @@ TORRENT_VERSION_NAMESPACE_2
 #endif
 	};
 
-	// This alert is posted every time an outgoing peer connect attempts succeeds.
+	// This alert is posted every time an incoming or outgoing peer connect attempts succeeds.
 	struct TORRENT_EXPORT peer_connect_alert final : peer_alert
 	{
+		enum class direction_t { in, out };
+
 		// internal
 		TORRENT_UNEXPORT peer_connect_alert(aux::stack_allocator& alloc, torrent_handle h
-			, tcp::endpoint const& ep, peer_id const& peer_id, socket_type_t type);
+			, tcp::endpoint const& ep, peer_id const& peer_id, socket_type_t type, direction_t direction);
 
 		TORRENT_DEFINE_ALERT(peer_connect_alert, 23)
 
 		static constexpr alert_category_t static_category = alert_category::connect;
 		std::string message() const override;
 
+		// Tells you if the peer was incoming or outgoing
+		direction_t const direction;
```

Comment:
I think it's a mistake to have `const` members, I'm moving away from that in the alerts

Reply:
I think `peer_connect_alert::message()` (in `alert.cpp`) should be updated to print this field as well

Reply:
RE: `const`, do you think I should remove it for the direction field here, or is it something to keep just to stay in line with the rest of the alert? Or should I try to remove all the `const` from this alert?

Reply:
I think it's fine to remove the other `const` for this alert too